src/coldfront_plugin_api/serializers.py

```diff
@@ -23,28 +36,88 @@ def get_status(self, obj: Project) -> str:
+        model = Resource
+        fields = ["id", "name", "resource_type"]
+
+    id = serializers.IntegerField()
+    name = serializers.CharField(required=False)
+    resource_type = serializers.SerializerMethodField(required=False)
+
+    def get_resource_type(self, obj: Resource):
+        return obj.resource_type.name
+
+
 class AllocationSerializer(serializers.ModelSerializer):
     class Meta:
         model = Allocation
-        fields = ["id", "project", "description", "resource", "status", "attributes"]
+        fields = ["id", "project", "description", "resources", "status", "attributes"]
 
-    resource = serializers.SerializerMethodField()
+    resources = ResourceSerializer(many=True)
     project = ProjectSerializer()
-    attributes = serializers.SerializerMethodField()
-    status = serializers.SerializerMethodField()
+    attributes = AllocationAttributeSerializer(
+        many=True, source="allocationattribute_set", required=False
+    )
+    status = serializers.SlugRelatedField(
+        slug_field="name", queryset=AllocationStatusChoice.objects.all()
+    )
+
+    # TODO (Quan): What about default start/end dates? Default quantity? Description? Justification?
+
+    def create(self, validated_data):
+        project_obj = Project.objects.get(id=validated_data["project"]["id"])
+        resource_obj = Resource.objects.get(id=validated_data["resources"][0]["id"])
+        allocation = Allocation.objects.create(
+            project=project_obj, status=validated_data["status"]
+        )
+        allocation.resources.add(resource_obj)
+        allocation.save()
 
-    def get_resource(self, obj: Allocation) -> dict:
-        resource = obj.resources.first()
-        return {"name": resource.name, "resource_type": resource.resource_type.name}
+        # TODO (Quan): If the status is `Active`, do we fire the `activate_allocation`
+        # signal as well to allow creating the project on remote cluster?
 
-    def get_attributes(self, obj: Allocation):
-        attrs = AllocationAttribute.objects.filter(allocation=obj)
-        return {
-            a.allocation_attribute_type.name: obj.get_attribute(
-                a.allocation_attribute_type.name
+        for attribute in validated_data.pop("allocationattribute_set", []):
+            AllocationAttribute.objects.create(
+                allocation=allocation,
+                allocation_attribute_type=attribute["allocation_attribute_type"],
+                value=attribute["value"],
             )
-            for a in attrs
-        }
 
-    def get_status(self, obj: Allocation) -> str:
-        return obj.status.name
+        logger.info(
+            f"Created allocation {allocation.id} for project {project_obj.title}"
+        )
+        return allocation
+
+    def update(self, allocation: Allocation, validated_data):
+        """Only allow updating allocation attributes for now"""
+        # TODO (Quan) Do we want to allow updating any other allocation properties?
```

Comment:
@joachimweyl @knikolla @naved001

Reply:
I'm not convinced about the appropriate way to handle attribute changes. When done through the web dashboard they create a `ChangeRequest`, whereas modified from here they would modify the attribute directly. This is a flow that we haven't really dealt with as of now. I would say for now disallow changing attributes until we have a clear use case and workflow.